packages/pangea-sdk/pangea_django/pangea_django_auth.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 Pangea Cyber Corporation
+# Author: Pangea Cyber Corporation
+
+from datetime import datetime
+import os
+import traceback
+
+from django.contrib.auth.backends import BaseBackend
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AnonymousUser
+
+from pangea.services.authn.authn import AuthN
+from pangea.config import PangeaConfig
+import pangea.services.authn.models as m
+
+UserModel = get_user_model()
+
+class PangeaAuthMiddleware():
+    class InvalidSessionException(Exception):
+        pass
+
+    def __init__(self, get_response):
+        self.authn = PangeaAuthentication()
+        self.get_response = get_response
+
+    def __call__(self, request):
+        active_token = None
+        try:
+            active_token = request.session.get("PANGEA_ACTIVE_TOKEN")
+            if active_token:
+                expires = active_token["expire"]
+                expires = datetime.fromisoformat(expires.rstrip("Z"))
```

Comment:
Why are we stripping the timezone? `fromisoformat` can parse zulu time

Reply:
apparently it chokes on it. i tried not stripping it